### ISSUE TITLE
Similar Files are collapsed (breaking react-bootstrap)

### ISSUE THREAD

**Issue by killroy42** [parcel-bundler/parcel#3523]
# 🐛 bug report

When two different modules contain the same code, they are collapsed such that the code does not get executed as required.

For example, react-bootstrap contains two files with this content:
```
import React from 'react';
export default React.createContext(null);
```

With parcel, these "two modules" actually execute only once and return the same value. If I add some comment to one (e.g. `//`) it works as expected.

## 🎛 Configuration (.babelrc, package.json, cli command)

Running parcel with:
```
parcel --no-cache ./src/index.html --public-url=/reacttest --no-hmr
```

## 🤔 Expected Behavior
Perhaps the code could be collapsed for space saving purposes, but it still needs to be executed individually.

## 😯 Current Behavior
Similar files are merged and eliminated and executed only once.

## 💁 Possible Solution
Ideally do not eliminate similar files, or ensure they are executed individually.

## 🔦 Context
This breaks basic behavior in react-bootstrap related to `CardContext` and `AccordionContext`.

**Comment by mischnic** [comment]
A duplicate of #3368.

**Comment by wojtekmaj** [comment]
I think it's severe enough to deserve a fix on 1.x branch.

**Comment by mischnic** [comment]
(The relevant function: 
https://github.com/parcel-bundler/parcel/blob/fe08980a6f256d94100d017e7c1f1e0b3a527ec0/packages/core/parcel-bundler/src/Asset.js#L227-L229
`this.name` or `this.relativeName` could also be factored in
)

**Comment by kalda341** [comment]
Agreed that this is worth fixing on 1.x. This issue has cost me several days in troubleshooting.
